extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -42,17 +43,26 @@ export abstract class AzureAuth implements vscode.Disposable {
 	protected readonly scopesString: string;
 	protected readonly clientId: string;
 	protected readonly resources: Resource[];
+	public authLibrary: string;
 
 
 	constructor(
 		protected readonly metadata: AzureAccountProviderMetadata,
+		protected readonly tokenCache: SimpleTokenCache,
 		protected readonly context: vscode.ExtensionContext,
 		protected clientApplication: PublicClientApplication,
 		protected readonly uriEventEmitter: vscode.EventEmitter<vscode.Uri>,
 		protected readonly authType: AzureAuthType,
-		public readonly userFriendlyName: string,
+		public readonly userFriendlyName: string
 	) {
-		this.clientApplication = clientApplication;
+		this.authLibrary = vscode.workspace.getConfiguration('azure').get('authenticationLibrary');
+		vscode.workspace.onDidChangeConfiguration((changeEvent) => {
+			const impactLibrary = changeEvent.affectsConfiguration('azure.authenticationLibrary');
```

Comment:
Do the libraries share tokens? Or is there any other state we need to worried about being cached incorrectly such that we'll need to clear out the token cache and/or reload ADS?

Reply:
The tokens are slightly different between the libraries but they have their own token caches, so they will be kept separate